woocommerce/woocommerce-ios
# Coupons: Update analytics for M2

### Description

This PR updates Tracks events for coupon edit and deletion following p91TBi-7vL-p2:
- Track new actions on the coupon details screen: "edit_tapped" and "delete_tapped"
- Track events when coupon deletion succeeds or fails.
- Track events when coupon update is initiated. Here we check to see which value is updated.
- Track events when coupon update succeeds or fails.

### Testing instructions

Build the app and look at the console for the Tracks event in Xcode console:
- Navigate to Menu > Coupons > select a coupon > select the ellipsis button.
- Select Edit Coupon, you should see an event in the console: `🔵 Tracked coupon_details` with one of the properties: `AnyHashable("action"): "tapped_edit"`.
- Edit any value of the coupon and then tap the Save button. You should see an event in the console: `🔵 Tracked coupon_update_initiated`. Look at the properties in that event, and make sure that they are all correct.
- If the update succeeds, you should see `🔵 Tracked coupon_update_success` or `🔵 Tracked coupon_update_failed` otherwise. In the case of failure, the error should also be logged.
- Back to the Coupon Details screen, select the ellipsis button and select "Delete Coupon". The console should log `🔵 Tracked coupon_details` with property `AnyHashable("action"): "tapped_delete"`.
- Proceed to delete the coupon - if the deletion succeeds you should see `🔵 Tracked coupon_delete_success`, otherwise `🔵 Tracked coupon_delete_failed` with an error.

### Screenshots

N/A

---
